test/gurka/test_landmarks.cc

```diff
@@ -327,55 +334,61 @@ TEST(LandmarkTest, TestTileStoreLandmarks) {
   GraphReader gr(landmark_map.config.get_child("mjolnir"));
   auto tile = gr.GetGraphTile(tile_id);
 
+  auto check_landmark = [&](const auto& landmark, const auto& point) {
+    auto ltype = static_cast<LandmarkType>((edge_id.id() + 1) % invalid_landmark);
+
+    const double rounding_error = 5e-7;
+    // we dont store this in the tile
+    EXPECT_EQ(landmark.id, 0);
+    // opposing edges will have a copy of the edge we associated it to so we need to check it
+    bool skip = false;
+    if (landmark.name != std::to_string(edge_id)) {
+      auto opp = gr.GetOpposingEdgeId(edge_id);
+      ASSERT_EQ(landmark.name, std::to_string(opp)) << "Unexpected landmark name or edge/opp_edge";
+      skip = true;
+    }
+    // we dont do deep check on opposing edge records
+    if (!skip) {
+      EXPECT_EQ(landmark.type, ltype);
+      EXPECT_NEAR(landmark.lng, point.first, rounding_error);
+      EXPECT_NEAR(landmark.lat, point.second, rounding_error);
+    }
+  };
+
   // we support up to 6 decimal precision for landmark lng/lat, so max rounding error is 5e-7
-  const double rounding_error = 5e-7;
 
+  edge_id.set_id(0);
   for (const auto& e : tile->GetDirectedEdges()) {
     auto ei = tile->edgeinfo(&e);
 
     // test EdgeInfo:GetTags
     auto tagged_values = ei.GetTags();
-
-    edge_index = 0;
     for (const auto& value : tagged_values) {
       if (value.first != baldr::TaggedValue::kLandmark)
         continue;
-
       Landmark landmark(value.second);
 
       // check data correctness
       std::vector<PointLL> shape = ei.shape();
       auto point = shape[shape.size() / 2];
-      auto ltype = static_cast<LandmarkType>((edge_index + 1) % invalid_landmark);
-
-      EXPECT_EQ(landmark.id, 0);
-      EXPECT_EQ(landmark.name, std::to_string(edge_index++));
```

Comment:
there was a bug here, incrementin the edge index but only if it got past the continue above